### ISSUE TITLE
Menu item tag filter for category blog doesn't reset on other blog views with no tags

### ISSUE THREAD

**Issue by qoolman21** [joomla/joomla-cms#18138]
### Steps to reproduce the issue
1. Create menu item of category (should contain some artcles with tags assigned) with blog layout and set filter by custom tag
2. Create menu item of category (should contain some artcles with no tags assigned) with blog layout and don't set filter by any tag (default setting)
3. Go to front view of category blog created in step 1.
4. Next go to front view of category blog created in step 2.

### Expected result
See articles based on menuitem settings in both category views

### Actual result
See no articles in category with no tags as tag filter remembers flter state form previously visited category with tag filter set in menuitem options 

### System information (as much as possible)
Any Joomla! 3.8

### Additional comments

Solution:
/components/com_content/models/articles.php

Original:
line 509: $tagId  = $this->getState('filter.tag');

Correct:
line 509: $tagId  = JFactory::getApplication()->input->get('filter_tag', 0, 'uint');

**Comment by ghost** [comment]
Can't confirm.

Setting both Menus as stated above. 

- In Menu "Blog and Tags" Articles are shown using Tag.
- In Menu "Blog without Tags" Articles are shown if they have a Tag or not. So setting Tag to "None" means "ignore if a Tag is set or not". Try: Set in Menu a not used Tag > no Article is shown.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18138">issues.joomla.org/tracker/joomla-cms/18138</a>.</sub>

**Comment by Hurmeli** [comment]
I can confirm.

Same exact problem after updating from 3.7.5 to 3.8.0.

This should be fixed ASAP as it renders category blog with tag filtering useless.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18138">issues.joomla.org/tracker/joomla-cms/18138</a>.</sub>

**Comment by ghost** [comment]
@Hurmeli can you make a Pull Request? This is needed as first Step to fix ASAP.

**Comment by mbabker** [comment]
The fix is not valid as it means this value cannot be passed through the model state but only through the request data (so it breaks other programmatic uses of the model, i.e. in modules).  By default, this particular state value is set from the request data in the same way for this model, so that alone is not the problem.

Category views use the category model on the frontend, and the category model uses the articles model to load data.  The category model uses a manner to populate this state value that also uses the active user session.  So the fix that is needed is to expand the state keys in a way that allows for multiple active contexts.  And this isn't a problem isolated to this particular view or configuration.

**Comment by csthomas** [comment]
I have not tested, line 117 in `populateState`, replace:
`$value = $this->getUserStateFromRequest($this->context . '.filter.tag', 'filter_tag', 0, 'int');`
by
`$value = $app->input->getInt('filter_tag', 0);`

**Comment by mbabker** [comment]
No.  Again, that removes being able to programmatically set the user state outside the model and forces the value to only be readable from the request.  Additionally, some state values are stored to the session so that when you navigate away from a filtered view you return with the same set of filters active.

Again, the problem is not the fact that the state is in use.  The problem is in the fact there is not enough uniqueness between state keys for the behavior to work consistently.

**Comment by csthomas** [comment]
```php
	protected function populateState($ordering = null, $direction = null)
	{
		$app = JFactory::getApplication('site');
		$pk  = $app->input->getInt('id');

		$this->setState('category.id', $pk);

		//$value = $this->getUserStateFromRequest($this->context . '.filter.tag', 'filter_tag', 0, 'int');
		$value = $app->input->getInt('filter_tag', 0);
		$this->setState('filter.tag', $value);
```

IMO for programmatically set we use `ignore_request` that do not call `populateState` in category model

**Comment by Hurmeli** [comment]
@franz-wohlkoenig
 
I'm not aware of any fix. I was simply confirming that the problem exists.

**Comment by mbabker** [comment]
Right, for programatic configuration you would pass that flag.  But for the default behavior when state is autopopulated, you are breaking more than you are fixing IMO by removing the ability to read and persist the value with the session, and again to fix that requires that the key that is used be more unique than what is basically right now `com_content.category.filter.tag`.  We have the PK a few lines above that, it is not hard to change the key to basically be `com_content.category.filter.$pk.tag` so that the persisted filter states do not apply to everything using the model but only to that specific category ID.  If you look later in the function, we are already doing this type of filtering with the menu item ID.

**Comment by csthomas** [comment]
OK, you solution is better. It looks like a search form that should remember the last searched fields per category.

**Comment by danielmreck** [comment]
I can also confirm this problem occurs in version 3.8.1.

**Comment by csthomas** [comment]
@qoolman21 @Hurmeli @danielmreck You can test patch at #18248

**Comment by ghost** [comment]
closed as having Pull Request #18248
